### PR TITLE
Reference Moist Air medium and compressor with fogging modeling

### LINKED ISSUE #503: [FEATURE] Compressor with ReferenceMoistAir medium
### What is your precise need ?

At the moment, the wet compression is not taken into consideration: the compressor uses FlueGases medium (a mix of gases and vapor). When a fogger is added at the inlet of the compressor, the computation of enthalpy and isentropic efficiency may be wrong.

### What solution do you see ?

The use of the medium ReferenceMoistAir (see https://doc.modelica.org/Modelica%204.0.0/Resources/helpDymola/Modelica_Media_Air_ReferenceMoistAir.html) is adapted for that case: it takes into consideration supersaturated humid air on a large range of pressure and temperature  (611.2 Pa ≤ p ≤ 10 MPa, 143.15 K ≤ T ≤ 2000 K)

### PR DESCRIPTION
**Pull request by nabily-met**
## Goal

- To be able to model wet compression, moist air should be used as medium for the `airCompressor` component. The existing `MoistAir` medium, can handle temperature up to ~370°C which is not enough for any of the compressors we model. The added `RefMoistAir` medium can handle up to 500°C, which is enough for all compressors, except GT26 model.
- Fogging example are also added.
- A converter from `RefMoistAir` to `FlueGases` is also added to split the compressor when the temperature is > 500°C

Resolves #503 

## Type of change 

- [ ] Bugfix
- [x] New feature
- [ ] Refactoring change
- [ ] Release & Version Update (don't forget to change the version number in `package.mo`)

Will it break anything in previous models ?

- [ ] Breaking change (If yes, make sure to point it out in the changelog)
- [x] Non-Breaking change

## Checklist

- [x] I have added the appropriate tags, reviewers, projects (and detailed the size and priority of my PR) and linked issues to this PR
- [x] I have performed a self-review of my own code
- [x] I have checked that all existing tests pass
- [ ] I have checked that my work is compatible with [OpenModelica](https://openmodelica.org/)
- [x] I have added/updated tests that prove my development works and does not break anything.
- [ ] I have made corresponding changes or additions to the documentation (in [Notion documentation](https://www.notion.so/metroscope/Metroscope-Modeling-Library-Documentation-MML3-WIP-50c8703c294446059d3b4a70d6ae4a71))
- [x] I have added corresponding entries to the [Changelog](../CHANGELOG.md)
- [x] I have checked for conflicts with target branch, and merged/rebased in consequence

You can also fill these out after creating the PR, but make sure to **check them all before submitting your PR for review**.

### CODE REVIEW

File: MetroscopeModelingLibrary/RefMoistAir.mo [hunk 1]
```diff
@@ -0,0 +1,634 @@
...
+    end HeatLoss;
+
+    model Leak
+
+      package RefMoistAirMedium = MetroscopeModelingLibrary.Utilities.Media.RefMoistAirMedium;
+      extends Partial.Pipes.Leak(
+        redeclare MetroscopeModelingLibrary.RefMoistAir.Connectors.Inlet C_in,
+        redeclare MetroscopeModelingLibrary.RefMoistAir.Connectors.Outlet C_out,
+        redeclare MetroscopeModelingLibrary.RefMoistAir.BaseClasses.IsoHFlowModel flow_model,
+        redeclare MetroscopeModelingLibrary.Sensors.MoistAir.FlowSensor flow_sensor,
```
Comment (hadrienp-met) [inline MetroscopeModelingLibrary/RefMoistAir.mo:296]:
```suggestion
        redeclare MetroscopeModelingLibrary.Sensors.RefMoistAir.FlowSensor flow_sensor,
```

**Review by hadrienp-met — CHANGES_REQUESTED** [review]
Be careful, you saved all the package `RefMoistAir` into one file only !

File: MetroscopeModelingLibrary/RefMoistAir/BaseClasses/FlowModel.mo [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  extends Partial.BaseClasses.FlowModel(
+    redeclare MetroscopeModelingLibrary.RefMoistAir.Connectors.Inlet C_in,
+    redeclare MetroscopeModelingLibrary.RefMoistAir.Connectors.Outlet C_out,
+    redeclare package Medium = RefMoistAirMedium) annotation (IconMap(primitivesVisible=false));
+
+  import MetroscopeModelingLibrary.Utilities.Units.Inputs;
+  Inputs.InputPower W_input(start=0);
+  Inputs.InputDifferentialPressure DP_input(start=0);
+equation
+  W = W_input;
```
Comment (hadrienp-met) [inline MetroscopeModelingLibrary/RefMoistAir/BaseClasses/FlowModel.mo:14]: Do you think we should add the relative_humidity here ?

File: MetroscopeModelingLibrary/RefMoistAir/Machines/AirCompressor.mo [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  // Failure modes
+  parameter Boolean faulty = false;
+  Units.Percentage eta_is_decrease(min = 0, max=100) "percentage decrease of eta_is";
+  Units.Percentage tau_decrease(min = 0, max=100) "percentage decrease of tau";
+
+  // Indicators
+  Units.MassFraction x_liq_in; // Liquid water mass fraction at the inlet
+  Units.MassFraction x_liq_out; // Liquid water mass fraction at the outlet
+  Real relative_humidity_in; // Relative humidity at the inlet
+  Real relative_humidity_out; // Relative humidity at the outlet
```
Comment (hadrienp-met) [inline MetroscopeModelingLibrary/RefMoistAir/Machines/AirCompressor.mo:31]: I think it's better to declare this in the flow model, like that we have access to the relative_humidity in all the components of RefMoistAir

**Review by hadrienp-met — APPROVED** [review]

File: MetroscopeModelingLibrary/RefMoistAir/Machines/AirCompressor.mo (hunk 3 above)
Comment (nabily-met) [reply MetroscopeModelingLibrary/RefMoistAir/Machines/AirCompressor.mo:31]: I will do that in a next PR

**Pull request merged**